lua/bigfile/init.lua

```diff
@@ -45,16 +45,29 @@ function M.is_feature_disabled(bufnr, feature_name)
   return false
 end
 
----@param filesize number File size in MiB
+---@param bufnr number
+---@return integer|nil size in MiB if buffer is valid, nil otherwise
+local function get_buf_size(bufnr)
+  local ok, stats = pcall(vim.loop.fs_stat, vim.api.nvim_buf_get_name(bufnr))
+  if not (ok and stats) then
+    return
+  end
+  return stats.size * 1024 * 1024
+end
+
+---@param bufnr number buffer id to match against
 ---@return feature[] features Features from rules that match the `filesize`
-local function match_features(filesize)
-  local MB = 1024 * 1024
+local function match_features(bufnr)
   local matched_features = {}
+  local filesize = get_buf_size(bufnr)
+  if not filesize then
+    return matched_features
+  end
   for _, rule in ipairs(config.rules) do
-    if filesize >= rule.size * MB then
+    if filesize >= rule.size then
 
       for _, raw_feature in ipairs(rule.features) do
-        table.insert(matched_features, features.get_feature(raw_feature))
+        table.insert(matched_features, features[raw_feature])
```

Comment:
```suggestion
        table.insert(matched_features, features.get_feature(raw_feature))
```
It's needed for validation and custom features